crates/extensions/az_mapper_ext/src/converter.rs

```diff
@@ -230,15 +260,20 @@ mod tests {
     #[test]
     fn converting_input_with_timestamp_produces_output_with_timestamp_given_add_timestamp_is_false()
     {
-        let mut converter = AzureConverter::new(false, Box::new(TestClock), MqttSchema::default());
+        let mut converter = AzureConverter::new(
+            false,
+            Box::new(TestClock),
+            MqttSchema::default(),
+            TimeFormat::Rfc3339,
+        );
 
         let input = r#"{
-            "time" : "2013-06-22T17:03:14.000+02:00",
+            "time" : "2013-06-22T17:03:14.001+02:00",
```

Comment:
Why this change has been required? Is this because of a precision-lost issue?

Reply:
I made the change originally because the timestamp was being parsed and then reinserted, which was altering the formatting of the timestamp. With the latest changes, this is no longer the case (we assume given the date is a string rather than a number that it doesn't need reformatting to RFC-3339, so we leave the value unaltered), so I'll undo the change.